### PR TITLE
Use Compute.Studio dev kit

### PR DESCRIPTION
**Pull request by jdebacker**
This PR updates the `compconfig` files to work with Compute.Studio.

### CODE REVIEW

**Comment by jdebacker** [comment]
@hdoupe, When I run `pytest` on the `cs_config` functions, I get:
```
(ccc-dev) A02655JDEBACKER:cs_config jason.debacker$ pytest
============================= test session starts ==============================
platform darwin -- Python 3.6.7, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /Users/jason.debacker/repos/CCC, inifile: pytest.ini
plugins: xdist-1.29.0, forked-1.0.2, pep8-1.0.6
collected 0 items / 1 errors

==================================== ERRORS ====================================
_________ ERROR collecting cs_config/cs_config/tests/test_functions.py _________
/anaconda3/envs/ccc-dev/lib/python3.6/site-packages/py/_path/local.py:701: in pyimport
    __import__(modname)
__init__.py:1: in <module>
    from .functions import *
functions.py:1: in <module>
    import ccc
../../ccc/__init__.py:4: in <module>
    from ccc.parameters import *
../../ccc/parameters.py:11: in <module>
    class Specification(paramtools.Parameters):
E   AttributeError: module 'paramtools' has no attribute 'Parameters'
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 2.67 seconds ============================
```

My `paramtools` seems up-to-date:
```
(ccc-dev) A02655JDEBACKER:cs_config jason.debacker$ conda list paramtools
# packages in environment at /anaconda3/envs/ccc-dev:
#
# Name                    Version                   Build  Channel
paramtools                0.10.1                     py_0    conda-forge
```

What else should I look at?

Thanks!

File: cs_config/cs_config/tests/test_functions.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import io
+from cs_config import functions
+
+
+class TestFunctions1(CoreTestFunctions):
+    get_version = functions.get_version
+    get_inputs = functions.get_inputs
+    validate_inputs = functions.validate_inputs
+    run_model = functions.run_model,
+    ok_adjustment = {"ccc": {"CIT_rate": 0.21}},
```
Comment (hdoupe) [inline cs_config/cs_config/tests/test_functions.py]:
```suggestion
    ok_adjustment = {"ccc": {"CIT_rate": 0.21}}
```

File: cs_config/cs_config/tests/test_functions.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import pandas as pd
+import io
+from cs_config import functions
+
+
+class TestFunctions1(CoreTestFunctions):
+    get_version = functions.get_version
+    get_inputs = functions.get_inputs
+    validate_inputs = functions.validate_inputs
+    run_model = functions.run_model,
```
Comment (hdoupe) [inline cs_config/cs_config/tests/test_functions.py]:
```suggestion
    run_model = functions.run_model
```

**Comment by hdoupe** [comment]
@jdebacker I'm not sure what the issue is. I was able to check out this branch and run the test suite. I removed my Cost-of-Capital-Calculator repo, removed the `ccc-dev` conda environment, and re-installed everything. I'm still unable to re-create the issue. How are you installing ccc when you run py.test on the `cs_config` package?

**Comment by jdebacker** [comment]
@hdoupe I think the issue was with ParamTools.  Conda is not working well for me and although it said version 0.10.1, I don't think it was.  I uninstalled and did `pip install paramtools`, which installed `paramtools 0.10.1` and I no longer go this error. 

Tests now failing with:

```
________________________ TestFunctions1.test_run_model _________________________

self = <cs_config.tests.test_functions.TestFunctions1 object at 0x1190b82e8>

    def test_run_model(self):
        self.test_all_data_specified()
>       inputs = self.get_inputs({})

/anaconda3/envs/ccc-dev/lib/python3.6/site-packages/cs_kit/validate.py:213:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

meta_params_dict = {}

    def get_inputs(meta_params_dict):
        '''
        Function to get user input parameters from COMP
        '''
        meta_params = MetaParams()
        meta_params.adjust(meta_params_dict)
        params = Specification()
        spec = params.specification(
            meta_data=True,
            serializable=True,
            year=meta_params.year
        )
        return {
             "meta_parameters": meta_params.dump(),
             "model_parameters": {
>                "ccc": spec.dump()
             }
         }
E       AttributeError: 'collections.OrderedDict' object has no attribute 'dump'

functions.py:54: AttributeError
===================== 3 failed, 3 passed in 13.01 seconds ======================
```
I thought `dump` was a method here of the `Parameters` class.  Ideas?

**Comment by jdebacker** [comment]
One error left.  Not sure of solution or what it means exactly:

```
============================= test session starts ==============================
platform darwin -- Python 3.6.7, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /Users/jason.debacker/repos/CCC, inifile: pytest.ini
plugins: xdist-1.29.0, forked-1.0.2, pep8-1.0.6
collected 6 items

tests/test_functions.py ....F.                                           [100%]

=================================== FAILURES ===================================
________________________ TestFunctions1.test_run_model _________________________

self = <cs_config.tests.test_functions.TestFunctions1 object at 0x110fa6e80>

    def test_run_model(self):
        self.test_all_data_specified()
        inputs = self.get_inputs({})
        check_get_inputs(inputs)

        class MetaParams(Parameters):
            defaults = inputs["meta_parameters"]

        mp_spec = MetaParams().specification(serializable=True)

        result = self.run_model(mp_spec, self.ok_adjustment)

>       assert cs_storage.LocalResult().load(result)

/anaconda3/envs/ccc-dev/lib/python3.6/site-packages/cs_kit/validate.py:223:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/anaconda3/envs/ccc-dev/lib/python3.6/site-packages/marshmallow/schema.py:720: in load
    data, many=many, partial=partial, unknown=unknown, postprocess=True
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <LocalResult(many=False)>
data = {'downloadable': [{'data': ',,Marginal Effective Total Tax Rate Under Baseline Policy,Marginal Effective Total Tax Rat... row6 col3" >0</td>\n            </tr>\n    </tbody></table>', 'media_type': 'table', 'title': 'METTR Summary Table'}]}

    def _do_load(
        self,
        data: typing.Mapping,
        *,
        many: bool = None,
        partial: typing.Union[bool, types.StrSequenceOrSet] = None,
        unknown: str = None,
        postprocess: bool = True
    ):
        """Deserialize `data`, returning the deserialized result.
        This method is private API.

        :param data: The data to deserialize.
        :param many: Whether to deserialize `data` as a collection. If `None`, the
            value for `self.many` is used.
        :param partial: Whether to validate required fields. If its
            value is an iterable, only fields listed in that iterable will be
            ignored will be allowed missing. If `True`, all fields will be allowed missing.
            If `None`, the value for `self.partial` is used.
        :param unknown: Whether to exclude, include, or raise an error for unknown
            fields in the data. Use `EXCLUDE`, `INCLUDE` or `RAISE`.
            If `None`, the value for `self.unknown` is used.
        :param postprocess: Whether to run post_load methods..
        :return: Deserialized data
        """
        error_store = ErrorStore()
        errors = {}  # type: typing.Dict[str, typing.List[str]]
        many = self.many if many is None else bool(many)
        unknown = unknown or self.unknown
        if partial is None:
            partial = self.partial
        # Run preprocessors
        if self._has_processors(PRE_LOAD):
            try:
                processed_data = self._invoke_load_processors(
                    PRE_LOAD, data, many=many, original_data=data, partial=partial
                )
            except ValidationError as err:
                errors = err.normalized_messages()
                result = (
                    None
                )  # type: typing.Optional[typing.Union[typing.List, typing.Dict]]
        else:
            processed_data = data
        if not errors:
            # Deserialize data
            result = self._deserialize(
                processed_data,
                error_store=error_store,
                many=many,
                partial=partial,
                unknown=unknown,
            )
            # Run field-level validation
            self._invoke_field_validators(
                error_store=error_store, data=result, many=many
            )
            # Run schema-level validation
            if self._has_processors(VALIDATES_SCHEMA):
                field_errors = bool(error_store.errors)
                self._invoke_schema_validators(
                    error_store=error_store,
                    pass_many=True,
                    data=result,
                    original_data=data,
                    many=many,
                    partial=partial,
                    field_errors=field_errors,
                )
                self._invoke_schema_validators(
                    error_store=error_store,
                    pass_many=False,
                    data=result,
                    original_data=data,
                    many=many,
                    partial=partial,
                    field_errors=field_errors,
                )
            errors = error_store.errors
            # Run post processors
            if not errors and postprocess and self._has_processors(POST_LOAD):
                try:
                    result = self._invoke_load_processors(
                        POST_LOAD,
                        result,
                        many=many,
                        original_data=data,
                        partial=partial,
                    )
                except ValidationError as err:
                    errors = err.normalized_messages()
        if errors:
            exc = ValidationError(errors, data=data, valid_data=result)
            self.handle_error(exc, data, many=many, partial=partial)
>           raise exc
E           marshmallow.exceptions.ValidationError: {'model_version': ['Unknown field.']}

/anaconda3/envs/ccc-dev/lib/python3.6/site-packages/marshmallow/schema.py:898: ValidationError
----------------------------- Captured stdout call -----------------------------
Output html...
===================== 1 failed, 5 passed in 32.51 seconds ======================
```

@hdoupe Suggestions?

**Comment by hdoupe** [comment]
@jdebacker that's a pretty gnarly stack trace. I'll work on that on the cs-kit side. It means that you have `model_version` in the outputs dictionary. You don't need that anymore since c/s can just grab it from the `get_version` function.

**Comment by jdebacker** [comment]
@hdoupe Thanks for the help.   All tests passing now and I think this is ready to go.  I'll do a separate PR to try to use the Tax-Calc params on Compute.Studio.  I'm having some trouble interpreting how Peter's doing this in his code.

**Comment by rickecon** [comment]
@jdebacker is this ready for merge?

**Comment by jdebacker** [comment]
@rickecon yes.

**Comment by jdebacker** [comment]
@hdoupe Actually, I think I got the TC params in.  If you can review this PR and then if it's good, I'll merge.  Look forward to these updates on Compute.Studio!

**Comment by jdebacker** [comment]
After merged, this PR will bring the CCC web app up-to-date with the latest Compute.Studio changes from [PR #3](https://github.com/compute-tooling/compute-studio-kit/pull/3)

**Comment by hdoupe** [comment]
@jdebacker thanks for making the updates! On a cursory glance, this looks good. I'm going out of town this weekend, but I will have time to give a more thorough review on Tuesday.

**Comment by hdoupe** [comment]
> @jdebacker thanks for making the updates! On a cursory glance, this looks good. I'm going out of town this weekend, but I will have time to give a more thorough review on Tuesday.

I'm on this today. Sorry for the delay!

**Comment by hdoupe** [comment]
@jdebacker I have this running locally and it looks good to me.

**Comment by jdebacker** [comment]
@hdoupe Thanks for confirming.  Merging this now.  Looking forward to the new CS features.

**Pull request merged**